shared/modules/provider-injection.js

```diff
@@ -77,15 +78,11 @@ function blockedDomainCheck() {
     'blueskybooking.com',
     'sharefile.com',
   ];
-  const currentUrl = window.location.href;
   let currentRegex;
   for (let i = 0; i < blockedDomains.length; i++) {
     const blockedDomain = blockedDomains[i].replace('.', '\\.');
-    currentRegex = new RegExp(
-      `(?:https?:\\/\\/)(?:(?!${blockedDomain}).)*$`,
-      'u',
-    );
-    if (!currentRegex.test(currentUrl)) {
+    currentRegex = new RegExp(`(?:https?:\\/\\/)${blockedDomain}.*$`, 'u');
```

Comment:
It looks like this would no longer match on subdomains of blocked sites. It'd also still return a false positive for `https://docs.google.com`.

Instead of adjusting the regex, I'd recommend comparing origins and paths directly. e.g. first look for an exact origin match. And if the blocklist entry has a path component, block anything under that path.